### DISCUSSION TITLE
Csv file

### DISCUSSION THREAD

**Discussion by nourriahi** [idaholab/moose#24283] [Q&A General] [answered]
Hello, 

I wanted to get a csv file that includes data for the whole domain (2D) for, I have got earlier a csv file that includes variations only on x axis but when I wanted to do so for both axis (all domain) I did not manage to : 
```
[VectorPostprocessors]
  [t_sampler]
    type = LineValueSampler
    variable = temperature
    start_point = '0 0 0'
    end_point = '0.00008 0.000005 0'
    num_points = 20
    sort_by = x y
```

Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

Right now you only have something along (0,0) -> (0.00008 0.000005)

If you have a 2D domain, you may consider this instead:
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementValueSampler.html
Which will sample every element
or
https://mooseframework.inl.gov/source/vectorpostprocessors/NodalValueSampler.html
for every node

We are in need of a PlaneValueSampler if you have time to build one and contribute it.

Guillaume
